Gems/Terrain/Code/Source/TerrainRenderer/TerrainClipmapManager.cpp

```diff
@@ -52,32 +48,12 @@ namespace Terrain
     
     bool TerrainClipmapManager::UpdateSrgIndices(AZ::Data::Instance<AZ::RPI::ShaderResourceGroup>& terrainSrg)
     {
-        const AZ::RHI::ShaderResourceGroupLayout* terrainSrgLayout = terrainSrg->GetLayout();
-
-        m_macroColorClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::MacroColorClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_macroColorClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::MacroColorClipmaps);
-        m_macroNormalClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::MacroNormalClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_macroNormalClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::MacroNormalClipmaps);
-        m_detailColorClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::DetailColorClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_detailColorClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::DetailColorClipmaps);
-        m_detailNormalClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::DetailNormalClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_detailNormalClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::DetailNormalClipmaps);
-        m_detailHeightClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::DetailHeightClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_detailHeightClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::DetailHeightClipmaps);
-        m_detailMiscClipmapsIndex = terrainSrgLayout->FindShaderInputImageIndex(AZ::Name(TerrainSrgInputs::DetailMiscClipmaps));
-        AZ_Error(TerrainClipmapManagerName, m_detailMiscClipmapsIndex.IsValid(), "Failed to find terrain srg input image %s.", TerrainSrgInputs::DetailMiscClipmaps);
-
-        m_clipmapDataIndex = terrainSrgLayout->FindShaderInputConstantIndex(AZ::Name(TerrainSrgInputs::ClipmapData));
-        AZ_Error(TerrainClipmapManagerName, m_clipmapDataIndex.IsValid(), "Failed to find terrain srg input constant %s.", TerrainSrgInputs::ClipmapData);
-
-
-        bool IndicesValid = m_macroColorClipmapsIndex.IsValid() && m_macroNormalClipmapsIndex.IsValid() &&
-            m_detailColorClipmapsIndex.IsValid() && m_detailNormalClipmapsIndex.IsValid() && m_detailHeightClipmapsIndex.IsValid() &&
-            m_detailMiscClipmapsIndex.IsValid() && m_clipmapDataIndex.IsValid();
-
-        m_clipmapsNeedUpdate = true;
+        for (uint32_t i = ClipmapName::MacroColor; i < ClipmapName::Count; ++i)
```

Comment:
nit: this is a little confusing. One option is to have an enum for First that overlaps with MacroColor, but honestly unless you're expecting there to be some Clipmaps before MacroColor that you don't want to process, just using `0` here is fine.